pandas/core/frame.py

```diff
@@ -5007,6 +5007,121 @@ def rename(
             errors=errors,
         )
 
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None = ...,
+        method: str | None = ...,
+        axis: Axis | None = ...,
+        inplace: Literal[False] = ...,
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> DataFrame:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None,
+        method: str | None,
+        axis: Axis | None,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        *,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None,
+        *,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        *,
+        method: str | None,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        *,
+        axis: Axis | None,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        *,
+        method: str | None,
+        axis: Axis | None,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None,
+        *,
+        axis: Axis | None,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None,
+        method: str | None,
+        *,
+        inplace: Literal[True],
+        limit: int | None = ...,
+        downcast: dict | str | None = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def fillna(
+        self,
+        value: Scalar | dict | Series | DataFrame | None = ...,
```

Comment:
you've typed `value` in the overloads, perhaps let's type it in the function signature too? (line 5128)

Reply:
I can modify it, but let's wait until someone has confirmed the typing of `value` first.

Reply:
@LarWong perhaps let's revert typing `value` here, that can be done separately (and we'll probably need to be more precise than `dict`), the rest looks good

Reply:
@MarcoGorelli Sorry, but I'm not sure what you mean by revert typing. Do you mean:
```
    @overload
    def fillna(
        self,
        value,
```
As in not typing `value`?

Reply:
Exactly

Reply:
@MarcoGorelli Ok done! The outputs from `mypy` are the same after modifications (see above).